htmd/projections/metriccoordinate.py

```diff
@@ -71,26 +86,26 @@ def project(self, mol):
         mol = mol.copy()
         if self.pbc:
             mol.wrap(self._centersel)
-        mol.align(trajalnsel, refmol=self._refmol, refsel=self._refalnsel)
+
+        if trajalnsel is not None and self._refmol is not None:
+            mol.align(trajalnsel, refmol=self._refmol, refsel=self._refalnsel)
 
         coords = np.concatenate((mol.coords[atomsel, 0, :], mol.coords[atomsel, 1, :], mol.coords[atomsel, 2, :]))
         return np.transpose(coords)  # I need to permute the dimensions here to put frames on the first
 
     def _getSelections(self, mol):
-        if self._pc_trajalnsel is not None and self._pc_atomsel is not None and self._pc_centersel is not None:
-            trajalnsel = self._pc_trajalnsel
-            atomsel = self._pc_atomsel
-            centersel = self._pc_centersel
-        else:
-            trajalnsel = mol.atomselect(self._trajalnsel)
-            atomsel = mol.atomselect(self._atomsel)
-            centersel = mol.atomselect(self._centersel)
-        if np.sum(trajalnsel) == 0:
-            raise NameError('Alignment selection resulted in 0 atoms.')
+        trajalnsel = None
+        if self._trajalnsel is not None:
+            trajalnsel = self._pc_trajalnsel if self._pc_trajalnsel is not None else mol.atomselect(self._trajalnsel)
+            if np.sum(trajalnsel) == 0:
+                raise RuntimeError('Alignment selection resulted in 0 atoms.')
+
+        atomsel = self._pc_atomsel if self._pc_atomsel is not None else mol.atomselect(self._atomsel)
+        centersel = self._pc_centersel if self._pc_centersel is not None else mol.atomselect(self._centersel)
```

Comment:
why you need at all the _pc_* variables?
If they are not computed they are None and if they are computed, they are computed, just make them self.trajalign, etc

Reply:
See previous comment about `precalculate` :)